src/transitions/transition.d.ts

```diff
@@ -0,0 +1,17 @@
+import { Omit } from '..';
+import { TransitionProps } from 'react-transition-group/Transition';
+
+
+export interface TransitionProps extends Omit<TransitionProps, 'timeout'> {
+  in?: TransitionProps['in'];
+  // FIXME: Missing from @types/react-transition-group/Transition
+  mountOnEnter?: boolean;
+  unmountOnExit?: boolean;
+}
+
+export type TransitionTimeout = TransitionProps['timeout'];
```

Comment:
It looks like you're using several different variants of what should be the same type in this PR:

- `TransitionProps['timeout']`
- `TransitionProps['timeout'] | 'auto'`
- `TransitionTimeout` (this type alias)
- `TransitionTimeout` | 'auto'

It seems like this type should just be defined as

```ts
export type TransitionTimeout = TransitionProps['timeout'] | 'auto';
```

and then you should use it everywhere instead of any of the other variants, no?